libopenage/renderer/camera/camera.h

```diff
@@ -96,6 +106,19 @@ class Camera {
 	 */
 	void move_rel(Eigen::Vector3f direction, float delta = 1.0f);
 
+
+	/**
+	 * Move the camera position in the direction of a given vector taking the
+	 * camera boundaries into account.
+	 *
+	 * @param direction Direction vector. Added to the current position.
+	 * @param delta Delta for controlling the amount by which the camera is moved. The
+	 *              value is multiplied with the directional vector before its applied to
+	 *              the positional vector.
+	 * @param camera_boundaries X and Z boundaries for the camera in the scene.
+	 */
+	void move_rel(Eigen::Vector3f direction, float delta, CameraBoundaries camera_boundaries);
```

Comment:
Arguments that are/should not be modified can be `const` reference:

```suggestion
	void move_rel(Eigen::Vector3f direction, float delta, const CameraBoundaries &camera_boundaries);
```